commit 8851c1b2e6f98797c218289152ac34d7af0313bb
Author: David Edmundson <kde@davidedmundson.co.uk>
Date:   Wed Nov 5 17:47:28 2014 +0100

    Remove uneeded code to stop alt+f4 killing the window
    
    This fixes ksmserver not being able to logout or shutdown correctly as
    it was unable to kill Plasma.
    
    Tests showed that pressing alt+f4 no longer kills the window anyway
    (possibly a change in kwin?)
    
    BUG: 338360
    REVIEW: 121004

diff --git a/shell/desktopview.cpp b/shell/desktopview.cpp
index c465201..ac50aed 100644
--- a/shell/desktopview.cpp
+++ b/shell/desktopview.cpp
@@ -186,15 +186,7 @@ bool DesktopView::event(QEvent *e)
             }
         }
 
-    } else if (e->type() == QEvent::Close) {
-        if (!ShellManager::s_standaloneOption) {
-            //prevent ALT+F4 from killing the shell
-            e->ignore();
-            return true;
-        }
-
-    //FIXME: this should *not* be needed
-    } else if (e->type() == QEvent::FocusIn) {
+    } else if (e->type() == QEvent::FocusIn) {     //FIXME: this should *not* be needed
         ensureWindowType();
 
     } else if (e->type() == QEvent::FocusOut) {
